src/components/calcite-alert/calcite-alert.tsx

```diff
@@ -190,49 +226,23 @@ export class CalciteAlert {
   /** the slotted alert link child element  */
   private alertLinkEl?: HTMLCalciteLinkElement;
 
-  /** map dismissal durations */
-  private autoDismissDurations = {
-    slow: 14000,
-    medium: 10000,
-    fast: 6000
-  };
-
-  /** map icon strings */
-  private iconDefaults = {
-    green: "checkCircle",
-    yellow: "exclamationMarkTriangle",
-    red: "exclamationMarkTriangle",
-    blue: "lightbulb"
-  };
-
-  // when an alert is opened or closed, update queue and determine active alert
-  @Listen("calciteAlertSync", { target: "window" })
-  alertSync(event: CustomEvent): void {
-    if (this.queue !== event.detail.queue) {
-      this.queue = event.detail.queue;
-    }
-    this.queueLength = this.queue.length;
-    this.determineActiveAlert();
-  }
+  /** the computed icon to render */
+  private requestedIcon?: string;
 
-  // when an alert is first registered, trigger a queue sync to get queue
-  @Listen("calciteAlertRegister", { target: "window" })
-  alertRegister(): void {
-    if (this.active && !this.queue.includes(this.el as HTMLCalciteAlertElement)) {
-      this.queued = true;
-      this.queue.push(this.el as HTMLCalciteAlertElement);
-    }
-    this.calciteAlertSync.emit({ queue: this.queue });
-    this.determineActiveAlert();
-  }
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
 
-  /** emit the opened alert and the queue */
-  private openAlert(): void {
-    setTimeout(() => (this.queued = false), 300);
-    this.calciteAlertOpen.emit({
-      el: this.el,
-      queue: this.queue
-    });
+  /** determine which alert is active */
+  private determineActiveAlert(): void {
+    if (this.queue?.[0] === this.el) {
+      this.openAlert();
+      if (this.autoDismiss) {
+        setTimeout(() => this.closeAlert(), DURATIONS[this.autoDismissDuration]);
```

Comment:
Is there any case where this timeout needs to be cleared?

Reply:
I don't believe so, this PR just moved the definitions to a resource file.